### PR TITLE
Frank's suggestions on NB 03, 04, 05 and 08

### LINKED ISSUE #64: Notebook 3
Notebook 3 has some very long code blocks. It would be better to break that down into pieces.

Also NB3 introduces the save/load function at the end. One way to reduce the size of blocks would be to use that functionality earlier and reuse previously saved discretizations etc.

### LINKED ISSUE #66: Notebook 4/5
I suggest a significant change in notebook 4 (may affect notebook 5 too). 

First I would like to avoid the usage of the MSM method 'coarse_grain' because that currently just calls a HMM estimation. We haven't implemented any real MSM-based coarse-graining, it is really a new estimation, and should better be accompanied by ITS, CK-test etc. So I suggest to postpone the computation of a coarse-grained transition matrix to notebook 5 and make notebook 5 part of the standard set of notebooks.

Notebook 4 can and should include PCCA (computation of metastable sets). We just can't use it to compute a coarse-grained transition matrix yet. But we can compute mfpts, equilibrium probabilities, representative structures etc.

Then I am missing TPT. TPT can be done on the metastable sets without having a coarse-grained transition matrix, because the flux can be coarse-grained. So I suggest that goes into notebook 4.

Generally NB4 is quite complex and has very extensive code cells. Think of breaking them into pieces, maybe break the whole notebook into two notebooks if needed.

Finally the visualization of the four Alanine dipeptide structures is not very useful. Maybe someone with an eye for molecular structures can revise that.

### LINKED ISSUE #67: Notebook 5
HMM stuff:
- The first part is easier to illustrate if we use the same lag time for the good and bad discretization, that way it shows that the good discretization passes the MSM CK-test while the bad one fails, while the HMM should work with both discretizations
- The three-state poor discretization example is nice. Show some reference line in the ITS plot that illustrates that the HMM recovers the true timescale obtained from a good discretization.
- I suggest to do a short version of the analysis notebook again with the HMM. Things like plotting the transition work, computing the TPT flux, computing mfpts etc. are now very easy to do. The same with visualizing structures. I suggest to fix the analysis notebook first and then just repeat those parts of the analysis that can be done in the same way. 
- Again avoid invoking MSM.coarse_grain. I was tempted to retire this functions, but we will likely replace the HMM estimation implementation by something else in the future.

### PR DESCRIPTION
**Pull request by thempel**
fixes #66
fixes #64
fixes #67 

deletes the pentapeptide example from NBs 03, 04 and 05.

Could somebody revise the ala2 visualization in NB4? Did not install nglview.

### CODE REVIEW

**Comment by thempel** [comment]
Not sure. You are referring to the CK-test plots. ps and 0.001 ns are the same, so those plots come out nicely. If I call `plot_cktest` without the `units` argument, it prints `steps` to the axis. With `units='ns', dt=.001`, the same function returns a plot with ns axis labels which are barely readable (0.025 ns ...). 

It seems that the `dt_traj` parameter of `bayesian_msm` is not at all passed to `cktest`, probably because `bayesian_msm.lag` is still in unit of steps, not in ns. So I'd suggest to leave it this way.

If I'm not wrong, the only implication the `dt_traj='0.001 ns'` has on notebook 04 is on the fluxes. (The ITS class does not have this argument, so passing it to the `plot_implied_timescales()` function is the only option.) I'll add the units to the flux plots.

File: notebooks/04.2-pcca-tpt.ipynb [hunk 1]
```diff
@@ -0,0 +1,827 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "solution2": "hidden"
+   },
+   "outputs": [],
+   "source": [
+    "feat = pyemma.coordinates.featurizer(pdb)\n",
+    "feat.add_distances(feat.select_Heavy())\n",
+    "data = pyemma.coordinates.load(files, features=feat)\n",
```
Comment (cwehmeyer) [inline notebooks/04.2-pcca-tpt.ipynb]:
```Python-Traceback
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-22-0f372de7d34b> in <module>()
      3 data = pyemma.coordinates.load(files, features=feat)
      4 
----> 5 tica = pyemma.load('nb4.pyemma', model_name='ala2tica_tica')
      6 tica_output = tica.transform(data)
      7 tica_all = np.concatenate(tica_output)

~/git/PyEMMA/pyemma/__init__.py in load(filename, model_name)
     56     """
     57     from ._base.serialization.serialization import SerializableMixIn
---> 58     return SerializableMixIn.load(file_name=filename, model_name=model_name)
     59 
     60 

~/git/PyEMMA/pyemma/_base/serialization/serialization.py in load(cls, file_name, model_name)
    275             raise NotImplementedError('This feature is only available on Python3. Consider upgrading.')
    276         from .h5file import H5File
--> 277         with H5File(file_name, model_name=model_name, mode='r') as f:
    278             return f.model
    279 

~/git/PyEMMA/pyemma/_base/serialization/h5file.py in __init__(self, file_name, model_name, mode)
     41         self._file = h5py.File(file_name, mode=mode)
     42         self._parent = self._file.require_group('pyemma')
---> 43         self._current_model_group = model_name
     44 
     45     def rename(self, old, new, overwrite=False):

~/git/PyEMMA/pyemma/_base/serialization/h5file.py in _current_model_group(self, model_name)
     97             # not existent and read only
     98             if model_name not in self._parent and self._file.mode == 'r':
---> 99                 raise ValueError('model_name "{n}" not found in file {f}'.format(n=model_name, f=self._file.name))
    100             self.__group = self._parent.require_group(model_name)
    101 

ValueError: model_name "ala2tica_tica" not found in file /
```

File: notebooks/04.1-msm-analysis.ipynb [hunk 2]
```diff
@@ -0,0 +1,750 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "solution2": "hidden"
+   },
+   "outputs": [],
+   "source": [
+    "cluster.save('nb4.pyemma', model_name='ala2tica_cluster', overwrite=True)\n",
```
[the remark this answers is not in the document]
Comment (thempel) [reply notebooks/04.1-msm-analysis.ipynb]: Thanks for testing. I can't reproduce this right now but had it once before. @marscher Any idea why this happens? Did I use this feature as it was supposed be to used, saving everything into the same file?

File: notebooks/04.2-pcca-tpt.ipynb (hunk 1 above)
Comment (thempel) [reply notebooks/04.2-pcca-tpt.ipynb]: That's a follow-up on the error posted above.

**Pull request merged**